etcd3/client.py

```diff
@@ -86,45 +86,102 @@ def __init__(self, alarm_type, member_id):
         self.member_id = member_id
 
 
+class EtcdTokenCallCredentials(grpc.AuthMetadataPlugin):
+    """Metadata wrapper for raw access token credentials."""
+
+    def __init__(self, access_token):
+        self._access_token = access_token
+
+    def __call__(self, context, callback):
+        metadata = (('token', self._access_token),)
+        callback(metadata, None)
+
+
 class Etcd3Client(object):
     def __init__(self, host='localhost', port=2379,
-                 ca_cert=None, cert_key=None, cert_cert=None, timeout=None):
+                 ca_cert=None, cert_key=None, cert_cert=None, timeout=None,
+                 user=None, password=None):
+
         self._url = '{host}:{port}'.format(host=host, port=port)
 
-        cert_params = [c is not None for c in (cert_cert, cert_key, ca_cert)]
-        if all(cert_params):
-            # all the cert parameters are set
-            credentials = self._get_secure_creds(ca_cert,
-                                                 cert_key,
-                                                 cert_cert)
-            self.uses_secure_channel = True
-            self.channel = grpc.secure_channel(self._url, credentials)
-        elif any(cert_params):
-            # some of the cert parameters are set
-            raise ValueError('the parameters cert_cert, cert_key and ca_cert '
-                             'must all be set to use a secure channel')
+        cert_params = [c is not None for c in (cert_cert, cert_key)]
+        if ca_cert is not None:
+            if all(cert_params):
+                # all the cert parameters are set
+                credentials = self._get_secure_creds(
+                    ca_cert,
+                    cert_key,
+                    cert_cert
+                )
+                self.uses_secure_channel = True
+                self.channel = grpc.secure_channel(self._url, credentials)
+            elif any(cert_params):
+                # some of the cert parameters are set
+                raise ValueError(
+                    'to use a secure channel ca_cert is required by itself, '
+                    'or cert_cert and cert_key must both be specified.')
+            else:
+                credentials = self._get_secure_creds(ca_cert, None, None)
+                self.uses_secure_channel = True
+                self.channel = grpc.secure_channel(self._url, credentials)
```

Comment:
Sent PR https://github.com/kragniz/python-etcd3/pull/196 for certificate handling changes.